internal/provider/network/resource_network_constructor.go

```diff
@@ -47,29 +47,15 @@ func (c *Constructor) Setup() util.Processors {
 					vlanID      = i.(int)
 				)
 				if vlanID != 0 {
-					networkType = builder.NetworkTypeVLAN
-					clusterNetworkName := c.Network.Labels[networkutils.KeyClusterNetworkLabel]
-					c.Network.Spec.Config = fmt.Sprintf(builder.NetworkVLANConfigTemplate, c.Network.Name, clusterNetworkName, vlanID)
-				} else {
-					networkType = builder.NetworkTypeCustom
+					networkType = string(networkutils.L2VlanNetwork)
+				} else if vlanID == 0 {
+					networkType = string(networkutils.UntaggedNetwork)
 				}
+				clusterNetworkName := c.Network.Labels[networkutils.KeyClusterNetworkLabel]
+				c.Network.Spec.Config = fmt.Sprintf(builder.NetworkVLANConfigTemplate, c.Network.Name, clusterNetworkName, vlanID)
+
 				c.Network.Labels[networkutils.KeyVlanLabel] = strconv.Itoa(vlanID)
```

Comment:
If the VLAN ID is 0, a label will also indicate the VLAN ID is 0. The behavior does not match the network controller's backend code (actually mutator). VLAN ID 0 is just an internal representation of an untagged network for the controller's convenience. We don't need this label since we already have an untagged label.

Reply:
Thanks for the suggestion. I update the code to use another way to keep cluster network name.